### PR TITLE
Make location types non-nullable

### PR DESCRIPTION
**Pull request by esoergel**
Wait on tests
@sravfeyn

### CODE REVIEW

File: corehq/apps/locations/migrations/0012_auto__add_field_locationtype_emergency_level__add_field_locationtype_u.py [hunk 1]
```diff
@@ -33,14 +33,16 @@ def backwards(self, orm):
...
             'domain': ('django.db.models.fields.CharField', [], {'max_length': '255', 'db_index': 'True'}),
             'emergency_level': ('django.db.models.fields.DecimalField', [], {'default': '0.5', 'max_digits': '10', 'decimal_places': '1'}),
             u'id': ('django.db.models.fields.AutoField', [], {'primary_key': 'True'}),
             'name': ('django.db.models.fields.CharField', [], {'max_length': '255'}),
             'overstock_threshold': ('django.db.models.fields.DecimalField', [], {'default': '3.0', 'max_digits': '10', 'decimal_places': '1'}),
             'parent_type': ('django.db.models.fields.related.ForeignKey', [], {'to': u"orm['locations.LocationType']", 'null': 'True'}),
-            'understock_threshold': ('django.db.models.fields.DecimalField', [], {'default': '1.5', 'max_digits': '10', 'decimal_places': '1'})
+            'understock_threshold': ('django.db.models.fields.DecimalField', [], {'default': '1.5', 'max_digits': '10', 'decimal_places': '1'}),
+            'shares_cases': ('django.db.models.fields.BooleanField', [], {'default': 'False'}),
+            'view_descendants': ('django.db.models.fields.BooleanField', [], {'default': 'False'})
```
Comment (czue) [inline corehq/apps/locations/migrations/0012_auto__add_field_locationtype_emergency_level__add_field_locationtype_u.py]: what's up with these? i would have expected to see a new migration file and no changes to old ones

**Comment by czue** [comment]
do we have to worry about things will null types on prod or have you already checked those?

**Comment by sravfeyn** [comment]
Fail looks real https://travis-ci.org/dimagi/commcare-hq/jobs/59594093#L1599

File: corehq/apps/locations/migrations/0012_auto__add_field_locationtype_emergency_level__add_field_locationtype_u.py (hunk 1 above)
Comment (esoergel) [inline corehq/apps/locations/migrations/0012_auto__add_field_locationtype_emergency_level__add_field_locationtype_u.py]: The old migration files were inconsistent, I think from that last-minute shuffling we did.  These fields appear in earlier migration files, but not on this one.  This means if you do `$ ./manage.py schemamigration --auto`, it compares the python models to the snapshot from the lastest migration files, and tries to re-add these columns.

**Comment by esoergel** [comment]
I checked those a while ago, but we should check again before migration.  Looks like there's one location without a type on `ilsgateway-test-3`, not sure how that happened.  Anyways, checking is really simple, just `SQLLocation.objects.filter(location_type=None)`.  I'm gonna mark as don't merge until I have time to get back to this.

**Comment by esoergel** [comment]
Should be g2m now.  I checked prod, staging, and india, and there are no locations without location types.  If I'm still online when this is merged, I can run the migration from the preindex directory just in case.
@sravfeyn @czue

**Pull request merged**